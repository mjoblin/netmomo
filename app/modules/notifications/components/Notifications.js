import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { notification } from 'antd';

import appModule from 'AppRoot/modules/app';


class Notifications extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            connectedToShifty: props.shiftyConnectionStatus ===
                appModule.constants.SHIFTY_CONNECTED,
        };
    }

    componentWillReceiveProps(nextProps) {
        const connectedToShifty = nextProps.shiftyConnectionStatus ===
            appModule.constants.SHIFTY_CONNECTED;

        if (connectedToShifty !== this.state.connectedToShifty) {
            if (connectedToShifty) {
                notification.success({
                    message: 'Connected to shifty',
                    description: 'The shifty connection has been established.',
                });
            } else {
                notification.warning({
                    message: 'Disconnected from shifty',
                    description: 'The connection to shifty has been lost.',
                });

            }

            this.setState({ ...this.state, connectedToShifty });
        }
    }

    render() {
        return (null);
    }
}

Notifications.propTypes = {
    shiftyConnectionStatus: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
    shiftyConnectionStatus: appModule.selectors.shiftyConnectionStatus(state),
});

export default connect(
    mapStateToProps
)(Notifications);

