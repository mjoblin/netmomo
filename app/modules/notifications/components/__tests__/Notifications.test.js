import React from 'react';
import { shallow } from 'enzyme';
import { Notifications } from '../Notifications';

describe('Notifications', () => {
    it('matches snapshot', () => {
        const component = shallow(
            <Notifications
                hubConnectionStatus={'Connected'}
            />
        );

        expect(component).toMatchSnapshot();
    });
});