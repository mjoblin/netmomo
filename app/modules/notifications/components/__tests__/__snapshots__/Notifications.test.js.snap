// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Notifications
    hubConnectionStatus="Connected"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
